src/steadystate.jl

```diff
@@ -212,48 +212,34 @@ or
 # Parameters
 - `H::QuantumObject`: The Hamiltonian or the Liouvillian of the system.
 - `ψ0::QuantumObject`: The initial state of the system.
-- `tspan::Real=Inf`: The final time step for the steady state problem.
+- `tmax::Real=Inf`: The final time step for the steady state problem.
 - `c_ops::Union{Nothing,AbstractVector,Tuple}=nothing`: The list of the collapse operators.
 - `solver::SteadyStateODESolver=SteadyStateODESolver()`: see [`SteadyStateODESolver`](@ref) for more details.
 - `reltol::Real=1.0e-8`: Relative tolerance in steady state terminate condition and solver adaptive timestepping.
 - `abstol::Real=1.0e-10`: Absolute tolerance in steady state terminate condition and solver adaptive timestepping.
 - `kwargs...`: The keyword arguments for the ODEProblem.
 """
 function steadystate(
-    H::QuantumObject{MT1,HOpType},
-    ψ0::QuantumObject{<:AbstractArray{T2},StateOpType},
-    tspan::Real = Inf,
+    H::QuantumObject{DT1,HOpType},
+    ψ0::QuantumObject{DT2,StateOpType},
+    tmax::Real = Inf,
     c_ops::Union{Nothing,AbstractVector,Tuple} = nothing;
     solver::SteadyStateODESolver = SteadyStateODESolver(),
     reltol::Real = 1.0e-8,
     abstol::Real = 1.0e-10,
     kwargs...,
 ) where {
-    MT1<:AbstractMatrix,
-    T2,
+    DT1,
+    DT2,
     HOpType<:Union{OperatorQuantumObject,SuperOperatorQuantumObject},
     StateOpType<:Union{KetQuantumObject,OperatorQuantumObject},
 }
-    (H.dims != ψ0.dims) && throw(DimensionMismatch("The two quantum objects are not of the same Hilbert dimension."))
-
-    N = prod(H.dims)
-    u0 = sparse_to_dense(_CType(ψ0), mat2vec(ket2dm(ψ0).data))
-
-    L = MatrixOperator(liouvillian(H, c_ops).data)
-
     ftype = _FType(ψ0)
-    prob = ODEProblem{true}(L, u0, (ftype(0), ftype(tspan))) # Convert tspan to support GPUs and avoid type instabilities for OrdinaryDiffEq.jl
-    sol = solve(
-        prob,
-        solver.alg;
-        callback = TerminateSteadyState(abstol, reltol, _steadystate_ode_condition),
-        reltol = reltol,
-        abstol = abstol,
-        kwargs...,
-    )
+    cb = TerminateSteadyState(abstol, reltol, _steadystate_ode_condition)
+    sol = mesolve(H, ψ0, [ftype(0), ftype(tmax)], c_ops, progress_bar = Val(false), callback = cb)
 
-    ρss = reshape(sol.u[end], N, N)
-    return QuantumObject(ρss, Operator, H.dims)
+    ρss = sol.states[end]
+    return ρss
```

Comment:
Is it necessary to go through `mesolve` routine ?
In this way, `sol.states` will save both the initial state and final state.

Reply:
We should avoid redundancies in the package. If there is something already implemented, we should use it. I can add a `saveat` kwarg (also in the previous implementation it was absent). The actual code called is simpler that what one would think. Thanks to multiple dispatch, the ProgressBar-e_ops callback is not even created, and `e_ops` is just an empty `Tuple` in the latest implementation. So I think that the compiled code is very similar to doing everything manually as before.

Reply:
ok